core/modules/loader/ServerTcpBase.cc

```diff
@@ -0,0 +1,618 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2018 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+
+
+// Class header
+#include "loader/ServerTcpBase.h"
+
+// System headers
+#include <iostream>
+#include <unistd.h>
+
+// Third-party headers
+
+
+// qserv headers
+#include "loader/CentralWorker.h"
+#include "loader/LoaderMsg.h"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+namespace {
+LOG_LOGGER _log = LOG_GET("lsst.qserv.loader.ServerTcpBase");
+
+const int testNewNodeName = 73; // TODO Get rid of this, possibly make NodeName member of ServerTCPBase
+unsigned int testNewNodeValuePairCount = 81;
+const int testOldNodeName = 42; // TODO Get rid of this, possibly make NodeName member of ServerTCPBase
+unsigned int testOldNodeKeyCount = 1231;
+}
+
+namespace lsst {
+namespace qserv {
+namespace loader {
+
+
+bool ServerTcpBase::writeData(AsioTcp::socket& socket, BufferUdp& data) {
+    while (data.getBytesLeftToRead() > 0) {
+        // Read cursor advances (manually in this case) as data is read from the buffer.
+        auto res = boost::asio::write(socket,
+                                      boost::asio::buffer(data.getReadCursor(), data.getBytesLeftToRead()));
+        data.advanceReadCursor(res);
+    }
+    return true;
+}
+
+
+uint32_t ServerTcpBase::getOurName() {
+    return (_centralWorker == nullptr) ? 0 : _centralWorker->getOurId();
+}
+
+
+bool ServerTcpBase::testConnect() {
+    try
+    {
+        LOGS(_log, LOG_LVL_INFO, "ServerTcpBase::testConnect 1");
+        boost::asio::io_context io_context;
+
+        AsioTcp::resolver resolver(io_context);
+        AsioTcp::resolver::results_type endpoints = resolver.resolve("127.0.0.1", std::to_string(_port));
+
+        AsioTcp::socket socket(io_context);
+        boost::asio::connect(socket, endpoints);
+
+
+        // Get name from server
+        BufferUdp data(500);
```

Comment:
Why the magic `500`? 
Should tis be configured or parameterized via a constant?